internal/provider/trust_provider_resource.go

```diff
@@ -564,6 +584,17 @@ func appendMatchRuleIfExists(matchRules []aembit.TrustProviderMatchRuleDTO, valu
 	return matchRules
 }
 
+func appendMatchRulesIfExists(matchRules []aembit.TrustProviderMatchRuleDTO, values []basetypes.StringValue, attrName string) []aembit.TrustProviderMatchRuleDTO {
+	if len(values) > 0 {
+		for _, value := range values {
+			matchRules = append(matchRules, aembit.TrustProviderMatchRuleDTO{
```

Comment:
why not to call the overloaded 'appendMatchRuleIfExists()' for a single value? Less code duplication